server/giphy.go

```diff
@@ -27,7 +29,8 @@ type giphySearchResult struct {
 }
 
 func (p *giphyProvider) getAttributionMessage() string {
-	return "Powered by Giphy"
+	siteURL := os.Getenv("MM_SERVICESETTINGS_SITEURL")
```

Comment:
A more idiomatic version is to access the MM config via the API : `p.API.GetConfig().ServiceSettings.SiteURL` 
(this line would disappear anyway if the file is served via the `/public` folder)

Reply:
Good to know, I am not sure why I didn't see that in the API docs but like you mention it was not necessary after change to using /public

Reply:
I had to search through the github repo for the Mattermost plugins documentation, this little gem of information is hidden in the [Best Practices page for server plugins](https://developers.mattermost.com/extend/plugins/server/best-practices/).